core/src/main/java/org/pentaho/di/core/logging/LoggingRegistry.java

```diff
@@ -86,6 +86,7 @@ public class LoggingRegistry {
   private int purgeTimerCount;
   /** Stat that counts the amount of Objects removed from registry map.**/
   private int purgedObjectCount;
+  private boolean purgeTimerScheduledAlready = false;
```

Comment:
Just thinking: would it be an overkill to use an AtomicBoolean here?

Reply:
@smmribeiro It is  singleton and we calling it from the kettle environment init.  Do you still think atomic variable will help ?